src/lib.rs

```diff
@@ -250,6 +254,22 @@ impl RegexBuilder {
         self.0.backtrack_limit = limit;
         self
     }
+
+    /// Set the approximate size limit of the compiled regular expression.
+    ///
+    /// This option is forwarded from the wrapped `regex` crate.
+    pub fn size_limit(&mut self, limit: usize) -> &mut Self {
```

Comment:
Can you add some text here that makes it clear that this will only apply to parts of the regex that get delegated, and because of that the actual used size might be size_limit * the number of delegated regexes?

Also, I'm thinking maybe we should add a prefix to this option, like `delegate_size_limit` and `delegate_dfa_size_limit`, what do you think? `size_limit` sounds very generic and maybe at some point we'd want to have a size limit for the VM/non-delegated parts (not sure).

Reply:
The first version of my patch had a `inner_size_limit` etc. Then I thought of the following:

From what I understand `compile_inner` gets [called at least once with a "cooked" regex that is somehow proportional to the size of the original fancy regex](https://github.com/untitaker/fancy-regex/blob/9d650aaf8aaa77e5e1c610ba93bad60c5cd6153d/src/lib.rs#L326).

This would imply to me that, while subsets of the input regex may individually run into this limit too, the `size_limit` applies to the entire regex as well.

lmk if that makes sense

Reply:
That's only in the special case where the entire regex can be delegated (if the regex doesn't use *any* fancy features). When there are fancy features like backrefs or look-arounds, only parts of the regex will be delegated. For some regexes, that might mean 3/4 delegates.

Reply:
ok I changed it to `delegate_*`